src/cfnlint/schemaManager.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+import platform
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+        self.boto3_sts = boto3.client('sts')
+        self.boto3_cfn = boto3.client('cloudformation')
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, name, registry_type):
+        account_id = self.boto3_sts.get_caller_identity().get('Account')
+        username = None
+        path_split = os.getcwd().split('/')
+        try:
+            index = path_split.index('Users')
+            username = path_split[index + 1]
+        except ValueError:
+            raise ValueError
+
+        is_windows = platform.system() == 'win32'
+
+        for region in self.regions:
+            path = self.create_path(is_windows, username, account_id, region, name)
+
+            if not os.path.isdir(path):
+                self.create_folder(path, name, registry_type)
+
+    def create_path(self, is_windows, username, account_id, region, name):
+        if is_windows:
+            path = os.path.join(r'C:\Users', username, 'AppData', 'cloudformation', account_id, region, name)
```

Comment:
I think there are better, more python methods, for getting a users home directory.  https://docs.python.org/3/library/os.path.html#os.path.expanduser and https://docs.python.org/3/library/pathlib.html#pathlib.Path.home